Sources/App/Models/Migrations/AppUser.Create.swift

```diff
@@ -0,0 +1,31 @@
+import Fluent
+
+extension AppUser {
+    struct Create: Migration {
+        private let entity = AppUser()
+
+        func prepare(on database: Database) -> EventLoopFuture<Void> {
+            database
+                .schema(schema)
+                .id()
+                .field(.email, .string, .required)
+                .field(.name, .string, .required)
+                .field(.hashedPassword, .string, .required)
+                .field(.shouldResetPassword, .bool, .required)
+                .field(.createdAt, .datetime)
+                .field(.updatedAt, .datetime)
+                .unique(on: .id, .email)
+                .create()
+                .flatMap {
+                    database.createIndex(for: .email, onTable: schema)
+                }
```

Comment:
I tend to use `.transform(to: database.createIndex(for: .email, onTable: schema))` and would love to hear you input on the two approaches 🤔

Reply:
@heidipuk that function takes an existing future and thus the work has already started. In order to ensure that the second task follows the first you have to use flatMap like this.

Reply:
arhh, yes, okey... I was under the impression that a transform would always follow the task it was chained to and hence it was safe to use. But sounds like it was more luck that it didn't fail for me then 😅